Sim/SimG4Common/src/SingleParticleGeneratorAction.cpp

```diff
@@ -0,0 +1,92 @@
+#include "SimG4Common/SingleParticleGeneratorAction.h"
+#include <CLHEP/Random/RandFlat.h>
+
+#include "G4PhysicalConstants.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4Event.hh"
+#include "G4ParticleGun.hh"
+#include "G4ParticleTable.hh"
+#include "G4ParticleDefinition.hh"
+#include "G4HEPEvtInterface.hh"
+#include "G4RunManager.hh"
+#include "G4PrimaryVertex.hh"
+#include "G4PrimaryParticle.hh"
+
+
+SingleParticleGeneratorAction::SingleParticleGeneratorAction()
+{
+	std::cout<<" SingleParticleGeneratorAction() "<<std::endl;
+	particle="geantino";
+	SetEnergyRange(1*GeV,1*TeV);
+	SetEtaRange(-5.,5.);
+	SetPhiRange();
+	SetVertexPosition(0,0,0);
+}
+	
+void SingleParticleGeneratorAction::GeneratePrimaries(G4Event* anEvent)
+{
+    std::cout<<" GeneratePrimaries() "<<std::endl;
+   
+    G4ParticleTable* particleTable = G4ParticleTable::GetParticleTable();
+    G4ParticleDefinition* particleDef = particleTable->FindParticle(particle);
+    std::cout<<" particle definition "<<particleDef<<"+++"<<particle<<std::endl;
+    G4double mass=0;
+    if (particle!="geantino")
+    	mass=particleDef->GetPDGMass();
+    // std::cout << "particle mass = "<<mass<<std::endl;
```

Comment:
please delete code that is no longer necessary (or in case of debug statements use the msg-service).